src/linear_samplers/block_row_rand_replace.jl

```diff
@@ -0,0 +1,47 @@
+# This code was Written by Nathaniel Pritchard
+"""
+    LinSysBlkRowReplace <: LinSysBlkRowSampler 
+
+A mutable structure with fields to handle randomly subset block sampling with replacement, also known as
+vanilla block randomized kaczmarz. 
+
+# Fields
+- `block_size::Int64` - Specifies the size of each block.
+- `block::Vector{Int64}` - The list of all the rows in each block.
+
+Calling `LinSysBlkRowReplace()` defaults to setting `block_size` to 2. The `sample`
+function will handle the re-initialization of the fields once the system is provided.
+"""
+mutable struct LinSysBlkRowReplace <: LinSysBlkRowSampler
+    block_size::Int64
+    block::Vector{Int64}
+end
+
+LinSysBlkRowReplace(block_size) = LinSysBlkRowReplace(block_size, Int64[])
+LinSysBlkRowReplace() = LinSysBlkRowReplace(2, Int64[])
```

Comment:
If block size is supplied and `block` is kept for memory considerations (pending the response to the above comment), then you should just be able to allocate the `block` to an empty `block_size` array, right?

For v0.2, you may want to consider some general suggestions for when things are allocated. E.g., things should be allocated in the constructor if possible, or things should be allocated when they are needed

Reply:
This is my plan for version 0.2. To be consistent with the other techniques I have implemented, but we need to figure out some way to preallocated as much as possible before a function call.